From 14b26f587d4a2f2f584577f306eb158d981354a4 Mon Sep 17 00:00:00 2001
From: Hitendra Prajapati <hprajapati@mvista.com>
Date: Mon, 29 Nov 2021 10:18:52 +0530
Subject: [PATCH] linux-user: remove host stime() syscall

stime() has been withdrawn from glibc

(12cbde1dae6f "Use clock_settime to implement stime; withdraw stime.")

Implement the target stime() syscall using host
clock_settime(CLOCK_REALTIME, ...) as it is done internally in glibc.

Upstream-Status: Backport [https://git.qemu.org/?p=qemu.git;a=commit;h=0f1f2d4596aee037d3ccbcf10592466daa54107f]
Buglink: https://bugs.launchpad.net/qemu/+bug/1852115
Reported-by: Cole Robinson <crobinso@redhat.com>
Signed-off-by: Laurent Vivier <laurent@vivier.eu>
Reviewed-by: Peter Maydell <peter.maydell@linaro.org>
Message-Id: <20191112142556.6335-1-laurent@vivier.eu>

Signed-off-by: Hitendra Prajapati <hprajapati@mvista.com>
---
 linux-user/syscall.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 83bfd259..fa099a40 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -8147,10 +8147,11 @@ abi_long do_syscall(void *cpu_env, int num, abi_long arg1,
 #ifdef TARGET_NR_stime /* not on alpha */
     case TARGET_NR_stime:
         {
-            time_t host_time;
-            if (get_user_sal(host_time, arg1))
-                goto efault;
-            ret = get_errno(stime(&host_time));
+	     struct timespec ts;
+	     ts.tv_nsec = 0;
+	     if (get_user_sal(ts.tv_sec, arg1))
+                 goto efault;
+	     ret = get_errno(clock_settime(CLOCK_REALTIME, &ts));
         }
         break;
 #endif
-- 
2.25.1

